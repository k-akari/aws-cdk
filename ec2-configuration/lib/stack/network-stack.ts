import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { VPC } from '../resource/vpc';
import { Subnet } from '../resource/subnet';
import { InternetGateway } from '../resource/internet-gateway';
import { RouteTable } from '../resource/route-table';
import { NetworkAcl } from '../resource/network-acl';

export class NetworkStack extends Stack {
  public readonly vpc: Vpc;
  public readonly subnet: Subnet;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new VPC(this);
    this.vpc = vpc.vpc;
    this.subnet = new Subnet(this, this.vpc);
    const internetGateway = new InternetGateway(this, this.vpc);
    new RouteTable(this, this.vpc, this.subnet, internetGateway.igw);
    new NetworkAcl(this, this.vpc, this.subnet);
  }
}
