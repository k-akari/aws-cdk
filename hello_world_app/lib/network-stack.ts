import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from './resource/vpc';
import { Subnet } from './resource/subnet';
import { InternetGateway } from './resource/internetGateway';
import { ElasticIp } from './resource/elasticIp';
import { NatGateway } from './resource/natGateway';
import { RouteTable } from './resource/routeTable';
import { NetworkAcl } from './resource/networkAcl';

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    // VPC
    const vpc = new Vpc();
    vpc.createResources(this);

    // Subnet
    const subnet = new Subnet(vpc.vpc);
    subnet.createResources(this);

    // Internet Gateway
    const internetGateway = new InternetGateway(vpc.vpc);
    internetGateway.createResources(this);

    // Elastic IP
    const elasticIp = new ElasticIp();
    elasticIp.createResources(this);

    // NAT Gateway
    const natGateway = new NatGateway(
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      elasticIp.ngw1a,
      elasticIp.ngw1c
    );
    natGateway.createResources(this);

    // Route Table
    const routeTable = new RouteTable(
      vpc.vpc,
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      subnet.subnetPrivate1a,
      subnet.subnetPrivate1c,
      internetGateway.igw,
      natGateway.ngw1a,
      natGateway.ngw1c
    );
    routeTable.createResources(this);

    // Network ACL
    const networkAcl = new NetworkAcl(
      vpc.vpc,
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      subnet.subnetPrivate1a,
      subnet.subnetPrivate1c
    );
    networkAcl.createResources(this);
  }
}
