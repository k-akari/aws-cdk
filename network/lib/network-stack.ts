import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from "aws-cdk-lib/aws-ec2";

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const vpc = new Vpc(this, "sample-vpc", {});
  }
}
