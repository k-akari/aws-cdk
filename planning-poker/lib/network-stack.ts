import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from './resource/vpc';
import { Subnet } from './resource/subnet';
import { InternetGateway } from './resource/internet-gateway';
import { ElasticIp } from './resource/elastic-ip';
import { NatGateway } from './resource/nat-gateway';
import { RouteTable } from './resource/route-table';
import { NetworkAcl } from './resource/network-acl';

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new Vpc(this);
    const subnet = new Subnet(this, vpc);
    const internetGateway = new InternetGateway(this, vpc);
    const elasticIp = new ElasticIp(this);
    const natGateway = new NatGateway(this, subnet, elasticIp);
    new RouteTable(this, vpc, subnet, internetGateway, natGateway);
    new NetworkAcl(this, vpc, subnet);
  }
}
