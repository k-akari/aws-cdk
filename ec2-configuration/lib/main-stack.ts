import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { NetworkStack } from './stack/network-stack';
import { SecurityGroupStack } from './stack/security-group-stack';
import { IamStack } from './stack/iam-stack';
import { ServerStack } from './stack/server-stack';
import { LoadBalancerStack } from './stack/load-balancer-stack';

export class MainStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const networkStack = new NetworkStack(scope, 'NetworkStack', {
      stackName: 'network-stack'
    });

    const securityGroupStack = new SecurityGroupStack(scope, 'SecurityGroupStack', networkStack.vpc, {
      stackName: 'security-group-stack'
    });

    const iamStack = new IamStack(scope, 'IamStack', {
      stackName: 'iam-stack'
    });

    const serverStack = new ServerStack(scope, 'ServerStack', networkStack.subnet.private1a, iamStack.iamRole.instanceProfile, securityGroupStack.sg.ec2, {
      stackName: 'server-stack'
    });

    new LoadBalancerStack(scope, 'LoadBalancerStack', networkStack.vpc, networkStack.subnet.private1a, securityGroupStack.sg.alb, serverStack.ec2.mainInstance, {
      stackName: 'load-balancer-stack'
    });
  }
}
