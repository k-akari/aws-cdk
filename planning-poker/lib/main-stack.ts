import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { NetworkStack } from './stack/network-stack';
import { SecurityGroupStack } from './stack/security-group-stack';
import { EcrStack } from './stack/ecr-stack';
import { DatabaseStack } from './stack/database-stack';
import { SecretsManagerStack } from './stack/secrets-manager-stack';

export class MainStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const networkStack = new NetworkStack(scope, 'NetworkStack', {
      stackName: 'network-stack'
    });

    const securityGroupStack = new SecurityGroupStack(scope, 'SecurityGroupStack', networkStack.vpc, {
      stackName: 'security-group-stack'
    });

    new EcrStack(scope, 'EcrStack', {
      stackName: 'ecr-stack'
    });

    const secretsManagerStack = new SecretsManagerStack(scope, 'SecretsManagerStack', {
      stackName: 'secrets-manager-stack'
    })

    new DatabaseStack(scope, 'DatabaseStack', networkStack.subnet, securityGroupStack.sg, secretsManagerStack.ssm, {
      stackName: 'database-stack'
    })
  }
}
