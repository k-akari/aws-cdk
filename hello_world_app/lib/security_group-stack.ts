import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { SecurityGroup } from './resource/securityGroup';
import { NetworkStack } from '../../network/lib/network-stack';

export class SecurityGroupStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const networkStack = new NetworkStack(scope, 'NetworkStack', {
      stackName: 'network-stack'
    });

    // Security Group
    const securityGroup = new SecurityGroup(networkStack.);
    securityGroup.createResources(this);
  }
}
