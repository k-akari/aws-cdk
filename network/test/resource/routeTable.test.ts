import { App } from 'aws-cdk-lib';
import { Template, Match } from 'aws-cdk-lib/assertions';
import * as Network from '../../lib/network-stack';

test('RouteTable', () => {
  const app = new App({
    context: {
      'serviceName': 'service',
      'envType': 'test'
    }
  });
  const stack = new Network.NetworkStack(app, 'NetworkStack');
  const template = Template.fromStack(stack);

  template.resourceCountIs('AWS::EC2::RouteTable', 3);
  template.hasResourceProperties('AWS::EC2::RouteTable', {
    VpcId: Match.objectLike({
      Ref: 'Vpc'
    }),
    Tags: [{ 'Key': 'Name', 'Value': 'service-test-rtb-public' }]
  });
  template.hasResourceProperties('AWS::EC2::RouteTable', {
    VpcId: Match.objectLike({
      Ref: 'Vpc'
    }),
    Tags: [{ 'Key': 'Name', 'Value': 'service-test-rtb-private-1a' }]
  });
  template.hasResourceProperties('AWS::EC2::RouteTable', {
    VpcId: Match.objectLike({
      Ref: 'Vpc'
    }),
    Tags: [{ 'Key': 'Name', 'Value': 'service-test-rtb-private-1c' }]
  });

  template.resourceCountIs('AWS::EC2::Route', 3);
  template.hasResourceProperties('AWS::EC2::Route', {
    RouteTableId: Match.objectLike({
      Ref: 'RouteTablePublic'
    }),
    DestinationCidrBlock: '0.0.0.0/0',
    GatewayId: Match.objectLike({
      Ref: 'InternetGateway'
    }),
  });
  template.hasResourceProperties('AWS::EC2::Route', {
    RouteTableId: Match.objectLike({
      Ref: 'RouteTablePrivate1a'
    }),
    DestinationCidrBlock: '0.0.0.0/0',
    NatGatewayId: Match.objectLike({
      Ref: 'NatGateway1a'
    }),
  });
  template.hasResourceProperties('AWS::EC2::Route', {
    RouteTableId: Match.objectLike({
      Ref: 'RouteTablePrivate1c'
    }),
    DestinationCidrBlock: '0.0.0.0/0',
    NatGatewayId: Match.objectLike({
      Ref: 'NatGateway1c'
    }),
  });

  template.resourceCountIs('AWS::EC2::SubnetRouteTableAssociation', 4);
});
