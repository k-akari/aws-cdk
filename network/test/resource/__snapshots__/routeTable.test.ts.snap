// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteTable 1`] = `[Function]`;
